crates/pallets/starknet/src/lib.rs

```diff
@@ -581,11 +581,17 @@ pub mod pallet {
                 !ContractClasses::<T>::contains_key(transaction.tx().class_hash().0),
                 Error::<T>::ClassHashAlreadyDeclared
             );
-            // Check if contract is deployed
-            ensure!(
-                ContractClassHashes::<T>::contains_key(transaction.tx().sender_address()),
-                Error::<T>::AccountNotDeployed
-            );
+
+            let tx_version = transaction.tx().version();
+
+            // Avoiding account checks for V0 declares as they don't need to originate from an existing account
+            if tx_version != TransactionVersion::ZERO {
+                // Check if contract is deployed
+                ensure!(
+                    ContractClassHashes::<T>::contains_key(transaction.tx().sender_address()),
+                    Error::<T>::AccountNotDeployed
+                );
+            }
```

Comment:
I think I did removed this check in #1627

Reply:
I after rebasing it will be removed automatically